chapter8.adoc

```diff
@@ -30,8 +69,64 @@ security manager using secure memory.
 When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
```

Comment:
Please pay attention to this comment

Reply:
When medbgen is 1 then there are no restrictions on the type of debug. For example, the debugger can gain run control and single step M-mode and change its register state. This does not mean that the debugger can do what it could not otherwise do such as writing to a ROM, or  changing a locked PMP.

Reply:
Exactly, your comments has to explicitly address in spec, because the Debug spec defines the behavior to bypass locked PMP. Thus, obviously "no restrictions" cannot be said.

Reply:
The debug specification allows for bypassing a locked PMP and it should be when M is under external debug. There is no  need to restrict anything that can be done in debug mode when external debug is enabled. Bypassing PMP is no worse than executing instructions of debuggers choice from the program buffer and having full run control on M-mode.

Reply:
There is no definition in debug spec to support the argument "The debug specification allows for bypassing a locked PMP and it should be when M is under external debug. "

Reply:
> There is no definition in debug spec to support the argument "The debug specification allows for bypassing a locked PMP and it should be when M is under external debug. "

Anything and everything that the debug specification allows to occur for M-mode external debug should be allowed when M-mode is under external debug. This includes single stepping M-mode to and skipping over the locking of the PMPs themself.

Reply:
>There is no definition in debug spec to support the argument "The debug specification allows for bypassing a locked PMP and it should be when M is under external debug. "
Sorry for confusion, what I mean is debug spec doesn't have definition to support  argument "....when M is under external debug", that is when sub-M mode is allowed to debug, relaxedpriv could also be set. 

That's why no restrictions" cannot be said. It has to restrict relaxedpriv.

Reply:
That is unrelated to this comment thread which is about medbgen=1. 
I agree that the "as if `aamvirtual` = 1 and `MPP` != `M-mode`" should be expanded "as if `aamvirtual` = 1 and `MPP`!=`M-mode` and `relaxedpriv`=0"

Reply:
Sorry for the deviation. 
Let me rephrase it:
even for the mdbgen=1, relaxedpriv has to be restricted.

>"Bypassing PMP is no worse than executing instructions of debuggers choice from the program buffer and having full run control on M-mode."

1. M-mode cannot bypass locked PMP
2. relaxedpriv affect program buffer as well

Overall, with `relaxedpriv` uncontrained give the external debugger the capability even beyond the M-mode

Reply:
I don't see motivation for doing that when medbgen is 1. When medbgen is 1 all M-mode resources are fair game. medbgen can be used to prevent the PMP from being locked by M-mode in the first place.

Reply:
There is lifecycle requirement to mandate it. Even for M-mode, there are potentially multiple actors. Locked PMP can never be bypassed.
![logical_sec_model_fig1](https://github.com/riscv/riscv-smmtt/assets/21162377/75d49f9c-6bad-44aa-b71a-c37e0efbb57e)